examples/Makefile-simulation

```diff
@@ -91,6 +91,19 @@ else
 COMMONCFLAGS                   += -DOPENTHREAD_CONFIG_NCP_UART_ENABLE=1
 endif # NCP_SPI == 1
 
+ifeq ($(OTNS),1)
+VIRTUAL_TIME ?= 1
+
+ifneq ($(VIRTUAL_TIME),1)
+$(error OTNS requires virtual time simulations)
```

Comment:
Should we move these checks to a `examples/simulation/openthread-core-simulation-config-check.h`, similar to [examples/platforms/cc2538/openthread-core-cc2538-config-check.h](https://github.com/openthread/openthread/blob/master/examples/platforms/cc2538/openthread-core-cc2538-config-check.h)?

This will help ensure that the checks are included even when a different build system is used.

Reply:
However we can not check `OPENTHREAD_SIMULATION_VIRTUAL_TIME` and `OPENTHREAD_SIMULATION_VIRTUAL_TIME_UART` in `openthread-core-simulation-config-check.h` because these 2 macros are defined in `platform-config.h`, which is not included by `openthread-core-config.h`. Thoughts?

Reply:
How about putting the check in `platform-config.h`?

Reply:
`platform-config.h` looks good to me. I originally putted the check there, but moved away.